indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/SeekableStreamIndexTaskRunner.java

```diff
@@ -498,34 +515,17 @@ public void run()
 
           SequenceMetadata sequenceToCheckpoint = null;
           for (OrderedPartitionableRecord<PartitionIdType, SequenceOffsetType> record : records) {
-
-
-            // for Kafka, the end offsets are exclusive, so skip it
-            if (isEndSequenceOffsetsExclusive() &&
-                createSequenceNumber(record.getSequenceNumber()).compareTo(
-                    createSequenceNumber(endOffsets.get(record.getPartitionId()))) >= 0) {
-              continue;
-            }
-
-            // for the first message we receive, check that we were given a message with a sequenceNumber that matches
-            // our expected starting sequenceNumber
-            if (!verifyInitialRecordAndSkipExclusivePartition(record)) {
-              continue;
-            }
+            final boolean shouldProcess = verifyRecordInRange(record.getPartitionId(), record.getSequenceNumber());
 
             log.trace(
-                "Got stream[%s] partition[%s] sequence[%s].",
+                "Got topic[%s] partition[%s] offset[%s], shouldProcess[%s].",
```

Comment:
I personally prefer `offset` over `sequence` because the former is more obviously a position to me, but am indifferent about whether `stream` or `topic`.